### DISCUSSION TITLE
User/Device Name?

### DISCUSSION THREAD

**Discussion by KrautHolg** [audiamus/BookLibConnect#79] [General]
How can you add additional usernames and devicenames?  It is all grayed out in my case.

Thanks for everything amd keep going!!!

**Comment by audiamus** [comment]
Are you talking about AAX Audio Converter? Or Book Lib Connect? Only the latter supports user names, wrapped into "profiles". To create a new profile in Book Lib Connect, click the [Profiles] button to open the Profiles dialog, then [Add new]. Note: The library window must be closed to enable this.

**Comment by KrautHolg** [comment]
Hello Audiamus, 

Thanks a lot for your reply!  I am sorry if I mistakenly posted in the wrong place.

I am talking about Book Lib Connect Version 0.11.0.0

When I click on the “Profiles” Button, a new window pops up where I can create a new profile.  When I do that another window pops up, where I can choose among ten different regions.  I could create ten profiles (one for each region) this way.  For the same region, I can’t create another profile with a different username as that option is grayed out.  I am asking this, because we are a big family of voracious readers and listeners and we all have our different accounts.  Right now we are not so much interested in the different regions, but in the option to create multiple profiles for the same region.  

Could you please tell me how I would go about doing that?

And thanks again for all your work.  I am doing my best to spread the word on what excellent apps you have created here.  

Best, 
KrautHolg

**Reply by audiamus** [reply]
I see. Indeed, creating another profile for the same region is currently not supported. I have to investigate what the reason for that is. There must be one, particularly as an explicit warning message is shown for overwriting an existing profile. 

_I'll move this discussion to the Book Lib Connect repo._

**Comment by audiamus** [comment]
Coming back to this, after digging into the code and testing myself: 

Creating a new profile in the same region but for a different account should work. The source code says that the uniqueness constraint is the combination of region and Amazon account ID. And also my test succeeded.

This means I was wrong here in my earlier assumption that it isn't supported. You'll get a warning message, though, that if you sign in with the same account for the same region, the previous profile will be overwritten. After that, the new login-URL will be created and the procedure continues normally. I don't see any options being disabled in the New Profile dialog in this case. But you have to observe the work-flow, and click the Open URL or Copy URL buttons, or click into the login URL text box. Any of these actions will enable the controls in the lower section of the dialog, in particular the text box to paste the Amazon-generated authorization URL back in.

**Reply by KrautHolg** [reply]
Thank you very much for your reply.  I did manage to create another profile for the same region and I tested it.  It seems to work just fine.  The only mystifying thing for me now are the names of the profiles, but they don't really matter.

**Reply by audiamus** [reply]
I see. The profile alias name is currently static, once you created it. However, I don't think the alias is a key, so theoretically it could be changed. I will have a look.
